numpyro/distributions/continuous.py

```diff
@@ -1044,6 +1044,109 @@ def infer_shapes(
         return batch_shape, event_shape
 
 
+class MultivariateStudentT(Distribution):
+    arg_constraints = {
+        "df": constraints.positive,
+        "loc": constraints.real_vector,
+        "scale_tril": constraints.lower_cholesky,
+    }
+    support = constraints.real_vector
+    reparametrized_params = ["df", "loc", "scale_tril"]
+
+    def __init__(
+        self,
+        df,
+        loc=0.0,
+        scale_tril=None,
+        validate_args=None,
+    ):
+        if jnp.ndim(loc) == 0:
+            (loc,) = promote_shapes(loc, shape=(1,))
+        batch_shape = lax.broadcast_shapes(
+            jnp.shape(df), jnp.shape(loc)[:-1], jnp.shape(scale_tril)[:-2]
+        )
+        (self.df,) = promote_shapes(jnp.array(df), shape=batch_shape)
+        (self.loc,) = promote_shapes(loc, shape=batch_shape + loc.shape[-1:])
+        (self.scale_tril,) = promote_shapes(
+            scale_tril, shape=batch_shape + scale_tril.shape[-2:]
+        )
+        event_shape = jnp.shape(self.scale_tril)[-1:]
+        self._chi2 = Chi2(self.df)
+        super(MultivariateStudentT, self).__init__(
+            batch_shape=batch_shape,
+            event_shape=event_shape,
+            validate_args=validate_args,
+        )
+
+    def sample(self, key, sample_shape=()):
+        assert is_prng_key(key)
+        key_normal, key_chi2 = random.split(key)
+        std_normal = random.normal(
+            key_normal,
+            shape=sample_shape + self.batch_shape + self.event_shape,
+        )
+        z = self._chi2.sample(key_chi2, sample_shape)
+        y = std_normal * jnp.expand_dims(jnp.sqrt(self.df / z), -1)
+        return self.loc + jnp.squeeze(
+            jnp.matmul(self.scale_tril, y[..., jnp.newaxis]), axis=-1
+        )
+
+    @validate_sample
+    def log_prob(self, value):
+        n = self.scale_tril.shape[-1]
+        sample_shape = value.shape[:-2] if self.batch_shape != () else value.shape[:-1]
+        a_shape = sample_shape + self.batch_shape + (n, n)
+        b_shape = sample_shape + self.batch_shape + (n,)
+        y = solve_triangular(
+            jnp.broadcast_to(self.scale_tril, a_shape),
+            jnp.broadcast_to(value - self.loc, b_shape),
+            lower=True,
+        )
```

Comment:
You can use `M = _batch_mahalanobis(self.scale_tril, value - self.loc)` for the term `jnp.power(y, 2).sum(-1)` below. It is more scalable and might fix the shape errors that you faced. The logic for `sample_shape` here is a bit confusing to me (e.g. why sample shape is changing when batch_shape is non-trivial,...)

Reply:
🤯 well, I wish I'd known about this function sooner.